src/solver/simulation/adequacy.cpp

```diff
@@ -379,7 +382,7 @@ void Adequacy::simulationEnd()
     if (!preproOnly && study.runtime->interconnectionsCount() > 0)
     {
         auto balance = retrieveBalance(study, variables);
-        ComputeFlowQuad(study, pProblemesHebdo[0], balance, pNbWeeks);
+        ComputeFlowQuad(study, pProblemesHebdo[0], balance, pNbWeeks, resultWriter);
```

Comment:
`resultWriter` won't be used since `pProblemeHebdo->TypeDOptimisation == OPTIMISATION_QUADRATIQUE` in this case in src/solver/optimisation/opt_optimisation_hebdo.cpp:61.

You could just pass `NullResultWriter` to `OPT_OptimisationHebdomadaire` here, so you don't need to pass it to `ComputeFlowQuad` and `RecalculDesEchangesMoyens`.